docs/reference/mapping/types/dense-vector.asciidoc

```diff
@@ -45,5 +45,137 @@ PUT my-index-000001/_doc/2
 }
 
 --------------------------------------------------
+<1> `dims` is a required parameter that defines the number of vector dimensions
 
-<1> dims – the number of dimensions in the vector, required parameter.
+NOTE:  Unlike most other data types, dense vectors are always single-valued.
+It is not possible to store multiple values in one `dense_vector` field.
+
+==== Indexing vectors for kNN search
+
+experimental[]
+
+_k-nearest neighbor search_ or _kNN_ is the task of finding the k nearest
+vectors to a query vector, as measured by a similarity metric.
+
+As mentioned above, `dense_vector` fields can be used to rank documents
+through <<query-dsl-script-score-query,`script_score` queries>>. This lets you
+perform a brute-force kNN search by scanning all documents and ranking them by
+similarity.
+
+In many cases, a brute-force kNN search is not efficient enough. For this
+reason, the `dense_vector` type supports indexing vectors into a specialized
+data structure to support fast kNN search. You can enable indexing by setting
+through the `index` parameter:
```

Comment:
Thanks, this is great content! I was purposefully quite concise here, because I didn't think the field type reference was the right place to explain this. What would you think about saving this content for the high-level guide, and making sure to link to it here? I am not worried that we'll forget to add the content, we have plenty of time before 8.0 and we're tracking it as a requirement.

Reply:
That makes sense to me. I'll go ahead and mark this as resolved.